Library/Source/XMLHttpRequest.cpp

```diff
@@ -130,8 +130,19 @@ namespace Babylon
     {
         if (m_responseType.empty() || m_responseType == XMLHttpRequestTypes::ResponseType::Text)
         {
-            return m_runtimeImpl.LoadUrlAsync<std::string>(m_url).then(arcana::inline_scheduler, m_runtimeImpl.Cancellation(), [this](const std::string& data) {
-                m_responseText = std::move(data);
+            return m_runtimeImpl.LoadUrlAsync<std::string>(m_url).then(arcana::inline_scheduler, m_runtimeImpl.Cancellation(), [this](const std::string& data)
+            {
+                // check UTF-8 BOM encoding
+                if (data.size() >= 3 && data[0] == int8_t(0xEF) && data[1] == int8_t(0xBB) && data[2] == int8_t(0xBF))
```

Comment:
what's the `int8_t`s for?

Reply:
unsigned char(string) -> char conversion(constant)

Reply:
I see. You can do `\xEF` instead to avoid this, but see my other comment. We probably shouldn't be checking this at all and instead use something to convert binary to text to cover all the cases.

Reply:
Good idea! I'll take a closer look at curl api